src/HealthChecks.AzureStorage/AzureBlobStorageHealthCheck.cs

```diff
@@ -7,49 +6,36 @@ namespace HealthChecks.AzureStorage
-
-        private readonly string? _connectionString;
-        private readonly string? _containerName;
-        private readonly BlobClientOptions? _clientOptions;
-
-        private static readonly ConcurrentDictionary<string, BlobServiceClient> _blobClientsHolder = new();
-
-        public AzureBlobStorageHealthCheck(string connectionString, string? containerName = default, BlobClientOptions? clientOptions = null)
-        {
-            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
-            _containerName = containerName;
-            _clientOptions = clientOptions;
-        }
-
-        public AzureBlobStorageHealthCheck(Uri blobServiceUri, TokenCredential credential, string? containerName = default, BlobClientOptions? clientOptions = null)
+        private readonly BlobServiceClient _blobServiceClient;
+        private readonly AzureBlobStorageHealthCheckOptions _options;
+
+        public AzureBlobStorageHealthCheck(string connectionString, string? containerName = default, BlobClientOptions? clientOptions = default)
+            : this(
+                  ClientCache.GetOrAdd(connectionString, k => new BlobServiceClient(k, clientOptions)),
+                  new AzureBlobStorageHealthCheckOptions { ContainerName = containerName })
+        { }
+
+        public AzureBlobStorageHealthCheck(Uri blobServiceUri, TokenCredential credential, string? containerName = default, BlobClientOptions? clientOptions = default)
+            : this(
+                  ClientCache.GetOrAdd(blobServiceUri?.ToString()!, _ => new BlobServiceClient(blobServiceUri, credential, clientOptions)),
+                  new AzureBlobStorageHealthCheckOptions { ContainerName = containerName })
+        { }
+
+        public AzureBlobStorageHealthCheck(BlobServiceClient blobServiceClient, AzureBlobStorageHealthCheckOptions options)
         {
-            _blobServiceUri = blobServiceUri ?? throw new ArgumentNullException(nameof(blobServiceUri));
-            _azureCredential = credential ?? throw new ArgumentNullException(nameof(credential));
-            _containerName = containerName;
-            _clientOptions = clientOptions;
+            _blobServiceClient = blobServiceClient ?? throw new ArgumentNullException(nameof(blobServiceClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
         }
 
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                var blobServiceClient = GetBlobServiceClient();
-                await foreach (var page in blobServiceClient.GetBlobContainersAsync(cancellationToken: cancellationToken).AsPages(pageSizeHint: 1))
-                {
-                    break;
-                }
+                await _blobServiceClient.GetPropertiesAsync(cancellationToken);

```

Comment:
This just broke #783 again

Reply:
Added new PR here: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/1409